spec/integration/outputs/index_spec.rb

```diff
@@ -0,0 +1,129 @@
+require "logstash/codecs/plain"
+require "logstash/event"
+require "json"
+
+describe "indexing against running AppSearch", :integration => true do
+
+  let(:engine_name) {
+    (0...10).map { ('a'..'z').to_a[rand(26)] }.join
+  }
+
+  let(:config) {
+    {
+      "api_key" => ENV['APPSEARCH_PRIVATE_KEY'],
+      "engine" => engine_name,
+      "url" => "http://appsearch:3002"
+    }
+  }
+
+  subject { LogStash::Outputs::ElasticAppSearch.new(config) }
+
+  before(:each) do
+    create_engine(engine_name, "http://appsearch:3002", ENV['APPSEARCH_PRIVATE_KEY'])
+  end
+
+  private
+  def create_engine(engine_name, host, api_key)
+    url = host + "/api/as/v1/engines"
+    resp = Faraday.post(url, "{\"name\": \"#{engine_name}\"}",
+      "Content-Type" => "application/json",
+      "Authorization" => "Bearer " + api_key)
+    expect(resp.status).to eq(200)
+  end
+
+  describe "search and private keys are configured" do
+    let(:api_key_settings) {
+      {
+        :private => ENV['APPSEARCH_PRIVATE_KEY'],
+        :search => ENV['APPSEARCH_SEARCH_KEY']
+      }
+    }
+
+    it "setup apikeys" do
+      expect(api_key_settings[:private]).to start_with("private-")
+      expect(api_key_settings[:search]).to start_with("search-")
+    end
+  end
+
+  describe "indexing" do
+
+    before do
+      subject.register
+    end
+
+    describe "single event" do
+      let(:event) {
+        LogStash::Event.new("message" => "an event to index")
+      }
+
+      it "should be indexed" do
+        subject.multi_receive([event])
+
+        response = wait_for_response(1) do
+          execute_search_call(engine_name)
+        end
+        expect(response.status).to eq(200)
+        parsed_resp = JSON.parse(response.body)
+        expect(parsed_resp["meta"]["page"]["total_results"]).to eq(1)
+        results = parsed_resp["results"]
+        expect(results[0]["message"]["raw"]).to eq("an event to index")
+      end
+    end
+
+    private
+    def wait_for_response(expected_docs_count = 1)
+      retry_count = 0
+      loop do
+        response = yield
+        if (400..499).include?(response.status)
+          return response
+        elsif (200..299).include?(response.status)
+          search_resp = JSON.parse(response.body)
+          if search_resp["meta"]["page"]["total_pages"] == expected_docs_count
+            return response
+          end
+        end
+        if retry_count >= 5
+          return response
+        end
+        sleep 1
+        retry_count += 1
+      end
+    end
+
+    private
+    def execute_search_call(engine_name)
+      url = config["url"] + "/api/as/v1/engines/#{engine_name}/search"
+      resp = Faraday.post(url, '{"query": "event"}',
+            "Content-Type" => "application/json",
+            "Authorization" => "Bearer " + config["api_key"])
+    end
+
+    describe "multiple events" do
+      let(:events) {
+        generate_events(200) #2 times the slice size used to batch
+      }
```

Comment:
Nitpick: Either use multiline `do`...`end`:

```suggestion
      let(:events) do
        generate_events(200) #2 times the slice size used to batch
      end
```

or single-line `{`..`}` block syntax

```suggestion
      let(:events) { generate_events(200) } # 2 times the slice size used to batch
```